# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: 798e05478220
# Full Hash: 798e05478220c229cdceef4f16a02d2b7343c567
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-10 03:50:31
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Add a logging module for AudioSink. r=alwu,media-playback-reviewers
#   
#   Depends on D133642
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D133643
# ==============================================================================

diff -r b92a638392da -r 798e05478220 dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Mon May 09 13:20:48 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Mon May 09 13:20:49 2022 +0000
@@ -7,9 +7,16 @@
 #include "AudioSinkWrapper.h"
 #include "AudioSink.h"
 #include "VideoUtils.h"
+#include "mozilla/Logging.h"
 #include "mozilla/Result.h"
 #include "nsPrintfCString.h"
 
+mozilla::LazyLogModule gAudioSinkWrapperLog("AudioSinkWrapper");
+#define LOG(...) \
+  MOZ_LOG(gAudioSinkWrapperLog, mozilla::LogLevel::Debug, (__VA_ARGS__));
+#define LOGV(...) \
+  MOZ_LOG(gAudioSinkWrapperLog, mozilla::LogLevel::Verbose, (__VA_ARGS__));
+
 namespace mozilla {
 
 using media::TimeUnit;
@@ -60,12 +67,16 @@
     MOZ_ASSERT(mAudioSink);
     // Rely on the audio sink to report playback position when it is not ended.
     pos = mAudioSink->GetPosition();
+    LOGV("%p: Getting position from the Audio Sink %lf", this, pos.ToSeconds());
   } else if (!mPlayStartTime.IsNull()) {
     // Calculate playback position using system clock if we are still playing.
     pos = GetVideoPosition(t);
+    LOGV("%p: Getting position from the system clock %lf", this,
+         pos.ToSeconds());
   } else {
     // Return how long we've played if we are not playing.
     pos = mPlayDuration;
+    LOGV("%p: Getting static position, not playing %lf", this, pos.ToSeconds());
   }
 
   if (aTimeStamp) {
@@ -130,6 +141,7 @@
 
 void AudioSinkWrapper::SetPlaying(bool aPlaying) {
   AssertOwnerThread();
+  LOG("%p: SetPlaying %s", this, aPlaying ? "true" : "false");
 
   // Resume/pause matters only when playback started.
   if (!mIsStarted) {
@@ -246,3 +258,6 @@
 }
 
 }  // namespace mozilla
+
+#undef LOG
+#undef LOGV
